compiler/rustc_monomorphize/src/mono_checks/abi_check.rs

```diff
@@ -111,6 +111,11 @@ fn wasm_abi_safe<'tcx>(tcx: TyCtxt<'tcx>, arg: &ArgAbi<'tcx, Ty<'tcx>>) -> bool
         }
     }
 
+    // Zero-sized types are dropped in both ABIs, so they're safe
+    if arg.layout.size.bytes() == 0 {
```

Comment:
Question: can this use `arg.layout.is_zst()` (does alignment matter here?)?

Reply:
Sorry I'm pretty unfamiliar with this code so I fear I probably am not the best to answer this. AFAIK any zero-sized-type will be omitted in the old/new ABI so I don't believe alignment matters, so I don't know which of these methods for testing for a zst is best.

Reply:
In that case, I think using `arg.layout.is_zst()` would be slightly better.